simpleui-server/src/SuiZMQ.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Created by Zack Beucler
+ * 
+ * potenitally add a send timeout ZMQ_SNDTIMEO
+ *                      http://api.zeromq.org/3-0:zmq-setsockopt
+ *      might be good to add because zmq will wait forever if not told otherwise
+ * 
+ * 
+ ****************************** NOTE ******************************
+ *
+ *  An http request can only request data from a single zmq socket
+ *
+ ******************************************************************
+ */
+
+
+var _zmq = require('zeromq');
+import {ServerUtil} from './server-util';
+import {Logger, LogLevel} from './server-logger';
+import { SuiData } from './sui_data';
+import { Queue } from './queue';
+
+
+export class ZMQ_Socker_Wrapper {
+    hostname: string;
+    timeout: number;
+    port: number;
+    socket: any;
+    http_queue: Queue;
+    
+
+    constructor(hostname: string='svcmachineapps', port: number, timeout: number=500) {
```

Comment:
I wouldn't have a default for hostname, in the general case there's not a sensible default value

Reply:
fixed